pkg/clustermesh/endpointslicesync/clustermesh.go

```diff
@@ -0,0 +1,194 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package endpointslicesync
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	"github.com/cilium/endpointslice-controller/endpointslice"
+	"k8s.io/client-go/informers"
+	cache "k8s.io/client-go/tools/cache"
+
+	"github.com/cilium/cilium/pkg/clustermesh/common"
+	cmtypes "github.com/cilium/cilium/pkg/clustermesh/types"
+	"github.com/cilium/cilium/pkg/clustermesh/wait"
+	"github.com/cilium/cilium/pkg/hive/cell"
+	"github.com/cilium/cilium/pkg/k8s"
+	"github.com/cilium/cilium/pkg/k8s/resource"
+	slim_corev1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/api/core/v1"
+	"github.com/cilium/cilium/pkg/kvstore/store"
+	"github.com/cilium/cilium/pkg/loadbalancer"
+	serviceStore "github.com/cilium/cilium/pkg/service/store"
+)
+
+// clusterMesh is a cache of multiple remote clusters
+type clusterMesh struct {
+	// common implements the common logic to connect to remote clusters.
+	common common.ClusterMesh
+
+	context       context.Context
+	contextCancel context.CancelFunc
+	Metrics       Metrics
+
+	// globalServices is a list of all global services. The datastructure
+	// is protected by its own mutex inside the structure.
+	globalServices *common.GlobalServiceCache
+
+	storeFactory store.Factory
+
+	concurrentClusterMeshEndpointSync int
+
+	meshPodInformer     *meshPodInformer
+	meshNodeInformer    *meshNodeInformer
+	meshServiceInformer *meshServiceInformer
+
+	endpointSliceMeshController  *endpointslice.Controller
+	endpointSliceInformerFactory informers.SharedInformerFactory
+}
+
+func newClusterMesh(lc cell.Lifecycle, params clusterMeshParams) *clusterMesh {
+	if !params.Clientset.IsEnabled() || params.ClusterMeshConfig == "" {
+		return nil
+	}
+
+	log.WithField("enabled", params.Cfg.ClusterMeshEnableEndpointSync).
```

Comment:
nit: this probably could be after checking if `ClusterMeshEnableEndpointSync` is enabled. 
I know it doesn't matter for helm deployments, but in case someone manually specifies flags etc it might make a difference.

Reply:
Hmm this was on purposed to log a `enabled=false` message even if it is disabled actually, but it might better to not do it?

Reply:
If `ClusterMeshConfig` is unspecified then we still don't log it - which would happen as it is only specified in case of `enableEndpointSliceSynchronization` being enabled in helm AFAIK.

Reply:
Ah yes good catch, then assuming that we should log even if it's not enabled this could be something like this:
`log.WithField("enabled", params.Cfg.ClusterMeshEnableEndpointSync && params.ClusterMeshConfig != "").` in the first line of the func/before any check happens? Or get rid of this enabled field and only log after all the early return checks are done? WDYT?

Reply:
I would probably log it only if it's enabled, but don't have a strong opinion here.